pkg/test/vpc_client/proxy.go

```diff
@@ -14,27 +14,22 @@ import (
 // LaunchProxyInstance will launch a proxy instance on the indicated zone.
 // If set imageID to empty, it will find the proxy image in the ProxyImageMap map
 // LaunchProxyInstance will return proxyInstance detail, privateIPAddress,CAcontent and error
-func (vpc *VPC) LaunchProxyInstance(imageID string, zone string, keypairName string, privateKeyPath string) (in types.Instance, privateIP string, proxyServerCA string, err error) {
-	var inst types.Instance
-	if imageID == "" {
-		var ok bool
-		imageID, ok = CON.ProxyImageMap[vpc.Region]
-		if !ok {
-			log.LogInfo("Cannot find proxy image of region %s in map ProxyImageMap, will copy from existing region", vpc.Region)
-			var err error
-			imageID, err = vpc.CopyImageToProxy(CON.ProxyName)
-			if err != nil {
-				log.LogError("Error to copy image ID %s: %s", imageID, err)
-				return inst, "", "", err
-			}
-			//Wait 30 minutes for image to active
-			result, err := vpc.WaitImageToActive(imageID, 30)
-			if err != nil || !result {
-				log.LogError("Error wait image %s to active %s", imageID, err)
-				return inst, "", "", err
-			}
-		}
+func (vpc *VPC) LaunchProxyInstance(zone string, keypairName string, privateKeyPath string) (inst types.Instance, privateIP string, proxyServerCA string, err error) {
+	filters := []map[string][]string{
+		{
+			"name": {
+				CON.PublicImageName,
+			},
+		},
+	}
+
+	output, err := vpc.AWSClient.DescribeImage([]string{}, filters...)
+	if err != nil || output == nil {
```

Comment:
Might be best to decompose these, as if the `err == nil` but the `output` is nil, then the log will not print the err correctly, and it will return an empty `types.Instance` along with a nil error, which will not be handled correctly